devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -0,0 +1,528 @@
+}
+
+def increaseHeatSetpoint() {
+	def currentMode = device.currentState("thermostatMode")?.value
+	if (currentMode != "off") {
+		float currentSetpoint = device.currentValue("heatingSetpoint")
+
+		currentSetpoint = currentSetpoint + setpointStep
+		setHeatingSetpoint(currentSetpoint)
+	}
+}
+
+def decreaseHeatSetpoint() {
+	def currentMode = device.currentState("thermostatMode")?.value
+	if (currentMode != "off") {
+		float currentSetpoint = device.currentValue("heatingSetpoint")
+
+		currentSetpoint = currentSetpoint - setpointStep
+		setHeatingSetpoint(currentSetpoint)
+	}
+}
+
+def setThermostatMode(value) {
+	log.debug "setThermostatMode($value)"
+	// Thermostat only supports heat
+}
+
+def heat() {
+	log.debug "heat"
+	// Thermostat only supports heat
+	//sendEvent("name":"thermostatMode", "value":"heat")
+}
+
+def configure() {
+	def requests = []
+	log.debug "binding to Thermostat cluster"
+
+	unschedule(scheduledUpdateWeather)
+	if (settings.zipcode) {
+		requests += updateWeather()
+		runEvery1Hour(scheduledUpdateWeather)
+	}
+
+	// This thermostat only supports heat
+	sendEvent("name":"thermostatMode", "value":"heat")
+
+	requests += delayBetween([
+			zigbee.addBinding(THERMOSTAT_CLUSTER),
+			// Configure Thermostat Cluster
+			zigbee.configureReporting(THERMOSTAT_CLUSTER, ATTRIBUTE_LOCAL_TEMP, DataType.INT16, 10, 60, 50),
+			zigbee.configureReporting(THERMOSTAT_CLUSTER, ATTRIBUTE_HEAT_SETPOINT, DataType.INT16, 1, 0, 50),
+			zigbee.configureReporting(THERMOSTAT_CLUSTER, ATTRIBUTE_PI_HEATING_STATE, DataType.UINT8, 300, 900, 5),
+
+			// Configure Thermostat Ui Conf Cluster
+			zigbee.configureReporting(THERMOSTAT_UI_CONFIG_CLUSTER, ATTRIBUTE_TEMP_DISP_MODE, DataType.ENUM8, 1, 0, 1),
+			zigbee.configureReporting(THERMOSTAT_UI_CONFIG_CLUSTER, ATTRIBUTE_KEYPAD_LOCKOUT, DataType.ENUM8, 1, 0, 1),
+
+			zigbee.configureReporting(zigbee.RELATIVE_HUMIDITY_CLUSTER, 0x0000, DataType.UINT16, 10, 300, 1),
+
+			// Read the configured variables
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_LOCAL_TEMP),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_HEAT_SETPOINT),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_PI_HEATING_STATE),
+			zigbee.readAttribute(THERMOSTAT_UI_CONFIG_CLUSTER, ATTRIBUTE_TEMP_DISP_MODE),
+			zigbee.readAttribute(THERMOSTAT_UI_CONFIG_CLUSTER, ATTRIBUTE_KEYPAD_LOCKOUT),
+			zigbee.readAttribute(zigbee.RELATIVE_HUMIDITY_CLUSTER, 0x0000)
+		], 200)
+
+	requests
+}
+
+// Unused Thermostat Capability commands
+def emergencyHeat() {
+	log.trace "${device.displayName} does not support emergency heat mode"
+}
+
+def cool() {
+	log.trace "${device.displayName} does not support cool mode"
+}
+
+def setCoolingSetpoint(degrees) {
+	log.trace "${device.displayName} does not support cool setpoint"
+}
+
+def on() {
+	heat()
+}
+
+def off() {
+	log.debug "off"
```

Comment:
Does it not support `off`? If not would be good to log that like the other unsupported commands